### DISCUSSION TITLE
customized key as file/note name instead of the citekey

### DISCUSSION THREAD

**Discussion by yayuwang0624** [emacs-citar/citar#647] [Q&A]
I am new to bibtex so not quite sure about whether it is the right direction but can we use, for example, the title field in one bibtex to create notes and library files?

I tried to hack the `citar-open-note-functions` to do so but I also noticed there are still a lot of cases in citar directly using citekey. Just wondering if there is a better solution or can we add something like a map function before using citekey as the display name/filepath or something?

**Comment by bdarcus** [comment]
Do you by chance use Org-Roam?

For the past six weeks or so, we've been working on a major architectural refactoring (see #634), which includes the note API.

Alongside that, I've been developing this:

https://github.com/emacs-citar/citar-org-roam/blob/main/citar-org-roam.el

That _might_ be the better place for this, since org-roam templates can configure the file paths and such, and in that the open function uses node-ids for finding the notes.

I should be merging that PR in the next 24-48 hours.

**Reply by yayuwang0624** [reply]
Thank you for the immediate reply.

I do use Org-Roam. `citar-org-roam` looks great and I will check it out.

Thanks for your contribution to this great package again.

**Reply by bdarcus** [reply]
Sounds good.

The only thing that needs fixing there is this:

https://github.com/emacs-citar/citar-org-roam/issues/2

**Reply by bdarcus** [reply]
OK, I've merged #634, FYI.
